src/udev/udev-builtin-net_id.c

```diff
@@ -566,6 +566,54 @@ static int pci_get_hotplug_slot(sd_device *dev, uint32_t *ret) {
         return -ENOENT;
 }
 
+static bool get_sun(sd_device *dev, uint32_t *ret) {
+        const char *attr;
+        int r;
+
+        assert(dev);
+        assert(ret);
+
+        r = device_get_sysattr_value_filtered(dev, "firmware_node/sun", &attr);
+        if (r < 0) {
+                log_device_debug_errno(dev, r, "Failed to read firmware_node/sun, ignoring: %m");
+                return false;
+        }
+
+        r = safe_atou32(attr, ret);
+        if (r < 0) {
+                log_device_warning_errno(dev, r, "Failed to parse firmware_node/sun (%s), ignoring: %m", attr);
+                return false;
+        }
+
+        return true;
+}
+
+static int pci_get_slot_from_firmware_node_sun(sd_device *dev, uint32_t *ret) {
+        int r;
+        sd_device *slot_dev;
+
+        assert(dev);
+        assert(ret);
+
+        /* Try getting the ACPI _SUN for the device */
+        if (get_sun(dev, ret))
+                return 0;
+
+        if ((r = sd_device_get_parent_with_subsystem_devtype(dev, "pci", NULL, &slot_dev)) < 0)
```

Comment:
we don't like this style that cramps a function call return value assignment and a check for it into one line. just do it in two, i.e.:

```c
r = sd_device…(…);
if (r < 0) 
        …
```